### PR TITLE
💧 Add `apollo-runtime-java`

### PR DESCRIPTION
**Pull request by martinbonnin**
First drop for `apollo-runtime-java`. Lots of things still missing but executing a simple query works 🎉 

See https://github.com/apollographql/apollo-kotlin/issues/4388

### CODE REVIEW

File: libraries/apollo-mockserver/build.gradle.kts [hunk 1]
```diff
@@ -19,7 +19,7 @@ kotlin {
         implementation(golatac.lib("atomicfu")) {
           because("We need locks for native (we don't use the gradle plugin rewrite)")
         }
-        implementation(golatac.lib("kotlinx.coroutines"))
+        api(golatac.lib("kotlinx.coroutines"))
```
Comment (martinbonnin) [inline libraries/apollo-mockserver/build.gradle.kts:22]: MockServer exposes `Flow` in its API

**Review by BoD — APPROVED** [review]
A few minor comments, but 👍

File: libraries/apollo-runtime-java/src/main/java/com/apollographql/apollo3/runtime/java/ApolloClient.java [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+        });
+      }
+
+      try {
+        Response response = callFactory.newCall(builder.build()).execute();
+        if (!response.isSuccessful()) {
+          ArrayList<HttpHeader> headers = new ArrayList<>();
+          response.headers().forEach(pair -> headers.add(new HttpHeader(pair.getFirst(), pair.getSecond())));
+
+          callback.onFailure(new ApolloHttpException(response.code(), headers, null, "HTTP error", null));
```
[the remark this answers is not in the document]
Comment (martinbonnin) [reply libraries/apollo-runtime-java/src/main/java/com/apollographql/apollo3/runtime/java/ApolloClient.java:124]: Yup, all the way down

File: libraries/apollo-runtime-java/src/main/java/com/apollographql/apollo3/runtime/java/ApolloClient.java [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+    return new DefaultApolloCall(this, operation);
+  }
+
+  public <D extends Query.Data> ApolloDisposable execute(ApolloRequest<D> request, ApolloCallback<D> callback) {
+    DefaultApolloDisposable disposable = new DefaultApolloDisposable();
+
+    ArrayList<ApolloInterceptor> interceptors = new ArrayList<>();
+    interceptors.add(new NetworkInterceptor(callFactory, serverUrl));
+
+    ApolloCallback<D> callbackWrapper = new ApolloCallback<D>() {
```
[the remark this answers is not in the document]
Comment (martinbonnin) [reply libraries/apollo-runtime-java/src/main/java/com/apollographql/apollo3/runtime/java/ApolloClient.java:68]: Not 100% sure TBH, I wanted to have something working but it's certainly not needed. I'll remove it.

File: libraries/apollo-runtime-java/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  id("apollo.library")
+}
+
+apolloLibrary {
+  javaModuleName("com.apollographql.apollo3.runtime.java")
+}
+
+repositories {
+  mavenCentral()
+}
```
[the remark this answers is not in the document]
Comment (martinbonnin) [reply libraries/apollo-runtime-java/build.gradle.kts:12]: Definitely don't need, thanks!

File: libraries/apollo-runtime-java/src/main/java/com/apollographql/apollo3/runtime/java/ApolloClient.java [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+        Response response = callFactory.newCall(builder.build()).execute();
+        if (!response.isSuccessful()) {
+          ArrayList<HttpHeader> headers = new ArrayList<>();
+          response.headers().forEach(pair -> headers.add(new HttpHeader(pair.getFirst(), pair.getSecond())));
+
+          callback.onFailure(new ApolloHttpException(response.code(), headers, null, "HTTP error", null));
+        } else {
+          BufferedSourceJsonReader jsonReader = new BufferedSourceJsonReader(response.body().source());
+          try {
+            ApolloResponse apolloResponse = Operations.parseJsonResponse(request.getOperation(), jsonReader, CustomScalarAdapters.Empty);
```
[the remark this answers is not in the document]
Comment (martinbonnin) [reply libraries/apollo-runtime-java/src/main/java/com/apollographql/apollo3/runtime/java/ApolloClient.java:128]: Yup, all the way down

**Pull request merged**